### PR TITLE
Make sinker delete only pods created by prow

### PR DESCRIPTION
**Pull request by kargakis**
Fixes https://github.com/kubernetes/test-infra/issues/4291

/assign spxtr

### CODE REVIEW

File: prow/cmd/sinker/main.go [hunk 1]
```diff
@@ -84,7 +84,8 @@ func clean(kc, pkc kubeClient, configAgent configAgent) {
 	}
 
 	// Now clean up old pods.
-	pods, err := pkc.ListPods(nil)
+	labels := map[string]string{kube.CreatedByProw: "true"}
+	pods, err := pkc.ListPods(labels)
```
Comment (kargakis) [inline prow/cmd/sinker/main.go]: @spxtr if you have lots of pods laying around, you will have to manually delete them, or manually label them
```
kubectl label pods -n POD_NS --all created-by-prow=true
```

Comment (kargakis) [reply prow/cmd/sinker/main.go]: Actually, this is a good candidate for including in the announcements section. Will update.

**Comment by spxtr** [comment]
/lgtm
Add to announcements?

**Comment by kargakis** [comment]
Yeah, waiting for https://github.com/kubernetes/test-infra/pull/4210 to get in first so I can rebase on top.

**Comment by kargakis** [comment]
The announcement section got merged by a different PR. Rebased and added an announcement section.

**Pull request merged**